app/src/main/kotlin/com/wire/android/notification/MessageNotificationManager.kt

```diff
@@ -72,41 +75,49 @@ class MessageNotificationManager @Inject constructor(private val context: Contex
-            appLogger.i("$TAG removing groupSummary")
-            notificationManager.cancel(NotificationConstants.MESSAGE_SUMMARY_ID)
-        } else if (oldData.size <= 1 && newData.size > 1) {
-            appLogger.i("$TAG adding groupSummary")
-            notificationManager.notify(NotificationConstants.MESSAGE_SUMMARY_ID, getSummaryNotification(userId))
-        }
-    }
+    private fun showSummaryIfNeeded(activeNotifications: Array<StatusBarNotification>) {
+        if (activeNotifications.find { it.id == NotificationConstants.MESSAGE_SUMMARY_ID } != null) return
 
-    private fun showConversationNotification(conversation: NotificationConversation, userId: String?) {
-        val notificationId = NotificationConstants.getConversationNotificationId(conversation.id)
-        val notification = getConversationNotification(conversation, userId)
-        notificationManager.notify(notificationId, notification)
+        appLogger.i("$TAG adding groupSummary")
+        notificationManager.notify(NotificationConstants.MESSAGE_SUMMARY_ID, getSummaryNotification())
     }
 
-    private fun hideNotification(conversationsId: ConversationId) =
-        notificationManager.cancel(NotificationConstants.getConversationNotificationId(conversationsId))
+    private fun showConversationNotification(
+        conversation: NotificationConversation,
+        userId: String?,
+        activeNotifications: Array<StatusBarNotification>
+    ) {
+        val notificationId = getConversationNotificationId(conversation.id)
+        getConversationNotification(conversation, userId, activeNotifications)?.let { notification ->
+            appLogger.i("$TAG adding ConversationNotification")
+            notificationManager.notify(notificationId, notification)
+        }
+    }
 
-    private fun getSummaryNotification(userId: String?) = NotificationCompat.Builder(context, NotificationConstants.MESSAGE_CHANNEL_ID)
+    private fun getSummaryNotification() = NotificationCompat.Builder(context, NotificationConstants.MESSAGE_CHANNEL_ID)
         .setSmallIcon(R.drawable.notification_icon_small)
         .setGroup(NotificationConstants.MESSAGE_GROUP_KEY)
         .setGroupSummary(true)
+        .setAutoCancel(true)
         .setDefaults(NotificationCompat.DEFAULT_ALL)
         .setPriority(NotificationCompat.PRIORITY_MAX)
         .setContentIntent(summaryMessagePendingIntent(context))
-        .setDeleteIntent(dismissSummaryPendingIntent(context, userId))
         .build()
 
-    private fun getConversationNotification(conversation: NotificationConversation, userId: String?) =
-        NotificationCompat.Builder(context, NotificationConstants.MESSAGE_CHANNEL_ID).apply {
+    /**
+     * Create or update existed [Notification]
+     * @return [Notification] for the conversation with all the messages in it (including the messages that been there before)
+     * OR null if there is no new messages in conversation and no need to update the existed notification.
+     */
```

Comment:
Small suggestion to improve the provided docs
```suggestion
    /**
     * Creates or updates existing [Notification]
     * @param [conversation] the notification object containing all the messages for the given conversation
     * @param [userId] the id of the user receiving the notifications
     * @param [activeNotifications] a list with the notifications that should be displayed to the user
     * @return [Notification] for the conversation with all the messages in it (including previous messages as well)
     * OR null if there is no new messages in conversation and no need to update the existed notification.
     */
```